tests/test_cart.py

```diff
@@ -1,15 +1,231 @@
+    cart.delete()
+
+    # test against new token
+    cart = get_or_create_anonymous_cart_from_token(uuid4())
+    assert Cart.objects.all().count() == 5
+    assert cart not in carts
+    assert cart.user is None
+    assert cart.status == Cart.OPEN
+    cart.delete()
+
+    # test against getting cart assigned to user
+    cart = get_or_create_anonymous_cart_from_token(opened_user_cart.token)
+    assert Cart.objects.all().count() == 5
+    assert cart not in carts
+    assert cart.user is None
+    assert cart.status == Cart.OPEN
+    cart.delete()
+
+
+def test_get_or_create_user_cart(test_user, opened_anonymous_cart,
+                                 cancelled_anonymous_cart, opened_user_cart,
+                                 cancelled_user_cart, admin_user):
+    cart = get_or_create_user_cart(test_user)
+    assert Cart.objects.all().count() == 4
+    assert cart == opened_user_cart
+
+    # test against getting closed carts
+    Cart.objects.create(user=admin_user, status=Cart.CANCELED)
+    queryset = Cart.objects.all()
+    carts = list(queryset)
+    cart = get_or_create_user_cart(admin_user)
+    assert Cart.objects.all().count() == 6
+    assert cart not in carts
+    assert cart.user is admin_user
+    assert cart.status == Cart.OPEN
+    cart.delete()
+
+
+def test_get_anonymous_cart_from_token(opened_anonymous_cart,
+                                       cancelled_anonymous_cart,
+                                       opened_user_cart,
+                                       cancelled_user_cart):
+    cart = get_anonymous_cart_from_token(opened_anonymous_cart.token)
+    assert Cart.objects.all().count() == 4
+    assert cart == opened_anonymous_cart
+
+    # test against getting closed carts
+    cart = get_anonymous_cart_from_token(cancelled_anonymous_cart.token)
+    assert Cart.objects.all().count() == 4
+    assert cart is None
+
+    # test against new token
+    cart = get_anonymous_cart_from_token(uuid4())
+    assert Cart.objects.all().count() == 4
+    assert cart is None
+
+    # test against getting cart assigned to user
+    cart = get_anonymous_cart_from_token(opened_user_cart.token)
+    assert Cart.objects.all().count() == 4
+    assert cart is None
+
+
+def test_get_user_cart(opened_anonymous_cart, cancelled_anonymous_cart,
+                       opened_user_cart, cancelled_user_cart, admin_user,
+                       test_user):
+    cart = get_user_cart(test_user)
+    assert Cart.objects.all().count() == 4
+    assert cart == opened_user_cart
+
+    # test against getting closed carts
+    Cart.objects.create(user=admin_user, status=Cart.CANCELED)
+    queryset = Cart.objects.all()
+    carts = list(queryset)
+    cart = get_user_cart(admin_user)
+    assert Cart.objects.all().count() == 5
+    assert cart is None
+
+
+def test_get_or_create_cart_from_request(monkeypatch, test_user):
+    token = uuid4()
+    queryset = Cart.objects.all()
+    request = Mock(spec=HttpRequest, user=test_user,
```

Comment:
py.test delivers `RequestFactory` as a fixture, so there's no need to use Mock to fake requests: http://pytest-django.readthedocs.io/en/latest/helpers.html#rf-requestfactory